Index: N:/extlibs/subversion-1.0.1/subversion/libsvn_wc/adm_files.c
===================================================================
--- N:/extlibs/subversion-1.0.1/subversion/libsvn_wc/adm_files.c	(revision 9261)
+++ N:/extlibs/subversion-1.0.1/subversion/libsvn_wc/adm_files.c	(working copy)
@@ -45,6 +45,28 @@
 /*** File names in the adm area. ***/
 
 
+char* svn_wc__adm_dir_name = 0;
+
+/* Retrieve the name of the administrative directory */
+const char* svn_wc_get_adm_dir_name ()
+{
+  if (!svn_wc__adm_dir_name)
+    svn_wc__adm_dir_name = strdup(".svn");
+
+  return svn_wc__adm_dir_name;
+}
+
+/* Set the name of the administrative directory. The
+ * function will make a copy of the passed in string */
+void svn_wc_set_adm_dir_name (const char* name)
+{
+  if (svn_wc__adm_dir_name)
+    free (svn_wc__adm_dir_name);
+
+  svn_wc__adm_dir_name = strdup (name);
+}
+
+
 /* Return the path to something in PATH's administrative area.
  * 
  * First, the adm subdir is appended to PATH as a component, then the
Index: N:/extlibs/subversion-1.0.1/subversion/include/svn_wc.h
===================================================================
--- N:/extlibs/subversion-1.0.1/subversion/include/svn_wc.h	(revision 9261)
+++ N:/extlibs/subversion-1.0.1/subversion/include/svn_wc.h	(working copy)
@@ -670,9 +670,15 @@
  * against importing administrative subdirs, so now the name is a
  * matter of public record.
  */
-#define SVN_WC_ADM_DIR_NAME   ".svn"
+#define SVN_WC_ADM_DIR_NAME   svn_wc_get_adm_dir_name()
 
+/* Retrieve the name of the administrative directory */
+const char* svn_wc_get_adm_dir_name();
 
+/* Set the name of the administrative directory. The
+ * function will make a copy of the passed in string */
+void svn_wc_set_adm_dir_name (const char* name);
+
 
 /* Entries and status. */
 
